jerry-core/api/jerry.c

```diff
@@ -764,6 +764,35 @@ jerry_value_is_undefined (const jerry_value_t value) /**< api value */
   return ecma_is_value_undefined (jerry_get_arg_value (value));
 } /* jerry_value_is_undefined */
 
+/**
+ * Perform the JavaScript 'typeof' operation.
+ *
+ * @return jerry_type_t value based on the 'typeof' operation result
+ */
+jerry_type_t
+jerry_value_typeof (const jerry_value_t value) /**< input value to check */
+{
+  jerry_assert_api_available ();
+
+  jerry_type_t return_type;
+  lit_magic_string_id_t lit_id = ecma_get_typeof_lit_id (jerry_get_arg_value (value));
+
+  JERRY_ASSERT (lit_id != LIT_MAGIC_STRING__EMPTY);
+
+  switch (lit_id)
+  {
+    case LIT_MAGIC_STRING_UNDEFINED: return_type = JERRY_TYPE_UNDEFINED;  break;
```

Comment:
What about JERRY_TYPE_NULL?

Reply:
As the ecma 262 5.1 reference describes the NULL in JS as object. However, I'm in favour of having a JERRY_TYPE_NULL, but then the typeof name could be a bit misleading.

Reply:
I see. Leave as it is.